packages/integration-tests/src/data_request.rs

```diff
@@ -67,8 +67,11 @@ fn post_data_request() {
     assert!(app.execute(Addr::unchecked(USER), cosmos_msg).is_err());
 
     // should be able to fetch data request
-    // TODO: this is ugly to loop through events, use Response.data once it's merged
-    let dr_id = &res.events.last().unwrap().attributes.last().unwrap().value;
+    // TODO: this is an ugly way to get the dr_id.
+    // although PostDataRequest on the DataRequest contract returns it in `data`, the Proxy contract does not yet.
+    // https://github.com/sedaprotocol/seda-chain-contracts/issues/68
+    let dr_id = &res.events.last().unwrap().attributes[2].value;
```

Comment:
I think the only way it gets cleaner is handling the error...

```rust
let dr_id = res.events
    .last()
    .and_then(|event| event.attributes.get(2))
    .map(|attr| &attr.value)
    .ok_or_else(|| SomeError)?;
    ```
    
    And then putting it in a function since we do this a few times.

Reply:
Moved this to a helper function, but I'll make it much cleaner shortly in https://github.com/sedaprotocol/seda-chain-contracts/issues/68 !